Mage/src/main/java/mage/game/GameState.java

```diff
@@ -853,6 +865,11 @@ public void addSimultaneousDamage(DamagedEvent damagedEvent, Game game) {
             event.addEvent(damagedEvent);
             addSimultaneousEvent(event, game);
         }
+        if (!isPermanentBatchUsed && isPermanentDamage) {
+            DamagedBatchEvent event = new DamagedBatchForOnePermanentEvent(damagedEvent.getTargetId());
```

Comment:
do you think these DamagedBatchForOne constructors should just take the first event and extract the target id from it?